src/components/modal-flow/ModalFlow.js

```diff
@@ -0,0 +1,373 @@
+            this.setState({ requestNextLayout: false, layout: requestNextLayout });
+            break;
+        case LAYOUT_TRANSITION.HALF_TO_FULL_EXITING:
+            this.prevLayout = layout;
+            this.setState({ layout: requestNextLayout });
+            break;
+        case LAYOUT_TRANSITION.HALF_TO_FULL_ENTERING:
+            this.isAnimatingLayout = false;
+
+            // Reset both `requestNextLayout` and `requestNextStepIndex` state variables since HALF_TO_FULL_ENTERING has ended
+            this.setState({
+                requestNextLayout: false,
+                requestNextStepIndex: false,
+                currentStepIndex: requestNextStepIndex,
+            });
+            break;
+        case LAYOUT_TRANSITION.HALF_BORDERED_TO_WIDE_EXITING:
+            this.prevLayout = layout;
+            this.setState({ layout: requestNextLayout });
+            break;
+        case LAYOUT_TRANSITION.HALF_BORDERED_TO_WIDE_ENTERING:
+            this.isAnimatingLayout = false;
+
+            // No need of resetting `requestNextStepIndex` as it will be reset later by `handleCurrentStepTransitionEnd`
+            this.setState({ requestNextLayout: false });
+            break;
+        case LAYOUT_TRANSITION.WIDE_TO_FULL_EXITING:
+            this.prevLayout = layout;
+            this.setState({ layout: requestNextLayout });
+            break;
+        case LAYOUT_TRANSITION.WIDE_TO_FULL_ENTERING:
+            this.isAnimatingLayout = false;
+
+            // Reset both `requestNextLayout` and `requestNextStepIndex` state variables since WIDE_TO_FULL_ENTERING has ended
+            this.setState({
+                requestNextLayout: false,
+                requestNextStepIndex: false,
+                currentStepIndex: requestNextStepIndex,
+            });
+            break;
+        case LAYOUT_TRANSITION.FULL_TO_HALF_EXITING:
+            this.prevLayout = layout;
+            this.setState({ layout: requestNextLayout });
+            break;
+        case LAYOUT_TRANSITION.FULL_TO_HALF_ENTERING:
+            this.isAnimatingLayout = false;
+
+            this.setState({
+                requestNextLayout: false,
+                requestNextStepIndex: false,
+                currentStepIndex: requestNextStepIndex,
+            });
+            break;
+        case LAYOUT_TRANSITION.FULL_TO_WIDE_EXITING:
+            this.prevLayout = layout;
+            this.setState({ layout: requestNextLayout });
+            break;
+        case LAYOUT_TRANSITION.FULL_TO_WIDE_ENTERING:
+            this.isAnimatingLayout = false;
+
+            this.setState({
+                requestNextLayout: false,
+            });
+            break;
+
+        default: break;
+        }
+    };
+}
+
+ModalFlow.propTypes = {
+    variant: PropTypes.oneOf(['simple', 'simple-with-feedback', 'advanced']),
+    step: PropTypes.string.isRequired,
+    children: PropTypes.node,
+};
+
+ModalFlow.defaultProps = {
+    variant: 'simple',
```

Comment:
I see that some components have a default variant, others do not. Question: should we standardize that?

Reply:
The only component that doesn't have a `default` variant is the `Button`. I'll update it for the sake of consistency.